code/builders/builder-vite/src/codegen-importfn-script.ts

```diff
@@ -21,34 +23,33 @@ function toImportPath(relativePath: string) {
 /**
  * This function takes an array of stories and creates a mapping between the stories' relative paths
  * to the working directory and their dynamic imports. The import is done in an asynchronous
- * function to delay loading. It then creates a function, `importFn(path)`, which resolves a path to
- * an import function and this is called by Storybook to fetch a story dynamically when needed.
+ * function to delay loading and to allow Vite to split the code into smaller chunks. It then
+ * creates a function, `importFn(path)`, which resolves a path to an import function and this is
+ * called by Storybook to fetch a story dynamically when needed.
  *
  * @param stories An array of absolute story paths.
  */
-async function toImportFn(stories: string[]) {
-  const { normalizePath } = await import('vite');
+export async function toImportFn(root: string, stories: string[]) {
   const objectEntries = stories.map((file) => {
-    const relativePath = normalizePath(relative(process.cwd(), file));
+    const relativePath = relative(root, file);
 
-    return `  '${toImportPath(relativePath)}': async () => import('/@fs/${file}')`;
+    return [toImportPath(relativePath), genDynamicImport(normalize(file))] as [string, string];
   });
 
-  return `
-    const importers = {
-      ${objectEntries.join(',\n')}
-    };
+  return dedent`
+    const importers = ${genObjectFromRawEntries(objectEntries)};
 
     export async function importFn(path) {
-        return importers[path]();
+      return await importers[path]();
```

Comment:
AFAIK this would only make a difference if there was a try/catch-block, which there isn't here. Or maybe I'm missing something?

Reply:
There is no noticeable difference (except that, in case of an exception, now the stacktrace also includes the importFn I believe). Personally, I prefer to always use "return await" because it's very easy to forget to add it once you add an exception handler. But I can revert this change if preferred.

Reply:
If this makes the stack trace better, I'm good with it! 👍